docs/auth_server.rst

```diff
@@ -0,0 +1,108 @@
+Authentication Server
+=====================
+
+Superdesk provides an OAuth2 authentication server for the production API. The server is
+using the `Client Credentials Grant`_ with `JWT authentication`_.
+
+The endpoint used in Superdesk is ``/api/auth_server/token``.
+
+.. _Client Credentials Grant: https://tools.ietf.org/html/rfc6749#section-4.4
+.. _JWT authentication: https://tools.ietf.org/html/rfc7519
+
+Allowing a client
+-----------------
+
+A client needs to be put in Superdesk's ``settings.py`` to be authorized to have an access
+token.
+
+The clients are put in the ``AUTH_SERVER_CLIENTS`` parameters which is a Python list
+of dictionaries. Each dictionary MUST have the following keys:
+
+======  ===============  ======================================================
+key     type             explanation
+======  ===============  ======================================================
+id      string           unique identifier of the client
+secret  string           shared secret between the client and the auth server
+scope   list of strings  list of allowed privileges for this client (see below)
+======  ===============  ======================================================
+
+Here is an example of client configuration:
+
+.. code:: python
+
+    AUTH_SERVER_CLIENTS = [
+        {
+            "id": "123456",
+            "secret": "789101112",
+            "scope": ["ARCHIVE_READ"]
+        }
+    ]
+
+
+Scope
+-----
+
+To interact with the production API, clients needs to be explicitly allowed to perform
+actions. This is done using the ``scope`` key in the settings, which is a list of
+well-known strings. The possible values are:
+
+=============  ======================================================
+permission     explanation
+=============  ======================================================
+ARCHIVE_READ   The client can read items from archive collection
+DESK_READ      The client can get desks metadata
+PLANNING_READ  The client can access planning, events and assignments
+CONTACT_READ   The client can get contacts metadata
+USERS_READ     The client can retrieve users
+=============  ======================================================
+
+Access token
+------------
+
+When created, the access token is stored in database in ``auth_server_token`` collection.
+You can use the service of the same name to access it.
+
+Please note that ``JWT`` does sign its payload but doesn't encrypt it, so you must be
```

Comment:
Why is the web token not encrypted? Let me know if I am incorrect, but isn't this token sent to the client so they can send this token with every request? The client only uses the token to send back to the server with every request to authorise the request. Therefor it would be best if this token was encrypted (as the token leaves your known network)

Reply:
The JWT token itself is used only between client and authorisation server, and there is not much sensitive data inside (client id mostly). The point of using it is stronger auth, and not passing the client secret on the wire. It's done via HTTPS anyway, so there is an encryption layer. It must mostly not be leaked to avoid replay attack, but there is protection against that (JTI).
The authorisation server then give an access token (again via HTTPS) which is a random series of bytes, this is the one which will then be used between the client and the resource owner. This one can be revoked.

In short: as long as HTTPS is OK, the token is OK. If HTTPS is broken, we have bigger problems anyway.